src/Admin/BaseFieldDescription.php

```diff
@@ -256,6 +261,56 @@ public function hasAssociationAdmin()
         return null !== $this->associationAdmin;
     }
 
+    final public static function clearFieldGetters()
+    {
+        self::$fieldGetters = [];
+    }
+
+    public function getFieldGetterKey($object, $fieldName)
+    {
+        return is_string($fieldName)
+            ? get_class($object).'-'.$fieldName
+                .(is_string($this->getOption('code'))
+                && '' !== $this->getOption('code')
+                    ? '-'.$this->getOption('code')
+                    : ''
+                )
+            : null;
+    }
+
+    public function hasCachedFieldGetter($object, $fieldName)
```

Comment:
Please make all new methods final

Reply:
Do we need them public? or can we make them protected or private?

Reply:
Declared them public to make them testable.
Otherwise they could be private, I think.

Reply:
If those private method are used by a public method you can test them testing that method instead of making them publics

Reply:
Well, that was the case before, as all `getFieldValue()` tests, the only really needed, passed :)
And then there's the new `clearFieldGetters()` which can be tested only if `hasCachedFieldGetter()` is public too..

Reply:
Hmm... I think you should be able to test all your methods only calling `getFieldValue` (maybe multiple times in a single test) and the `clearFieldGetters`

When @greg0ire asked for tests, he probably meant testing this specific caching behavior calling multiple times the `getFieldValue` to see if it really works as expected (returns cached value instead). Is that more or less right @greg0ire ?

Reply:
@jordisala1991 what I really want is to have coverage on each of these new lines, just to be sure everything still works.

Reply:
To be more precise, I cache only getter methods and their names, as the value itself might be dynamic.

Not sure how to test the new behavior with only calling `getFieldValue`. Measure time? :)
Or extract the caching stuff into another injectable service and then mock it to prove that cache methods are being called from within `getFieldValue`...

Reply:
Coveralls will tell us if the lines are covered or not, no need to prove anything ;)

Reply:
You don't need to measure time, you only need to make sure your test pass on all the code you write. Maybe I am wrong, but I don't see any reason for that methods to be public

Reply:
Coverallls is unhappy about private methods too :|